### ISSUE TITLE
Improve add_lines of Diff

### ISSUE THREAD

**Issue by arjunsinghy96** [coala/coala#4077]
`add_lines` raises ConflictError when adding lines after a line where `line_diff.add_after is not False` even if the lines to be added are same.
IMO ConflictError should be raised only when lines are different.

This issue needs discussion.

**Comment by meetmangukiya** [comment]
Right, when the group of lines being added at a given positions are same, then there is no uncertainty whether it should be added above the change or below the change, but to maintain consistency with other cases and to not make it more confusing I think we should raise a `ConflictError`.

However, this will be automatically fixed if https://github.com/coala/coala/issues/3885 is done. It is kinda related to that issue. But there is still discussion going on, on that issue.
